mlflow/tracking/_model_registry/client.py

```diff
@@ -172,12 +178,30 @@ def create_model_version(
                      :py:class:`mlflow.entities.model_registry.ModelVersionTag` objects.
         :param run_link: Link to the run from an MLflow tracking server that generated this model.
         :param description: Description of the version.
+        :param await_creation: Wait until the associated model version is finished being created.
         :return: Single :py:class:`mlflow.entities.model_registry.ModelVersion` object created by
                  backend.
         """
         tags = tags if tags else {}
         tags = [ModelVersionTag(key, str(value)) for key, value in tags.items()]
-        return self.store.create_model_version(name, source, run_id, tags, run_link, description)
+        mv = self.store.create_model_version(name, source, run_id, tags, run_link, description)
+        if await_creation:
+            max_datetime = datetime.utcnow() + timedelta(
+                seconds=AWAIT_MODEL_VERSION_CREATE_MAX_SLEEP_SECONDS
+            )
+            while mv.status == ModelVersionStatus.to_string(
+                ModelVersionStatus.PENDING_REGISTRATION
+            ):
+                if datetime.utcnow() > max_datetime:
+                    raise MlflowException(
```

Comment:
Perhaps rather than a boolean, I implemented `await_ready` as a numeric parameter that represents that amount of time to wait. By default, we can do 5 minutes but users can specify None if they don't want to wait and they can specify even more if they need it?

Reply:
Discussed offline, we're going to proceed with `await_creation_for` as a number representing seconds to wait. `0` and `None` will represent skipping waiting.

Reply:
The exception will still break people's existing workflows if the registration takes more than 5 minutes. Are we okay with that?

Reply:
I can think of a few options when the wait time runs out

1. throw an error
2. return model version but its still pending

In this case, I think we should throw an error because the user can specify how long they'd like to wait and it's consistent with other wait behaviors I've seen before so it's less surprising.

In contrast, if we return a model version that is still pending, the user still needs to write a check for pending because it can't trust this function to operate properly. In addition, if they don't check, they run the risk of it randomly failing in their workflow later on when they attempt to use it and it is still pending. Basically. this is a more surprising and less obvious behavior for the function. On the positive note, this is less likely to break existing users but this is experimental so I'm not inclined to worry about this potential issue because it will improves the user experience when it goes GA.

Reply:
OK let's just make sure to mark this PR as rn/breaking-change and clearly explain (in release notes) what to expect and do if you already have some workflow around it.

Reply:
I've added the rn/breaking-change and a clear explanation to the release notes.